app/jobs/github_commits_job.rb

```diff
@@ -1,63 +1,58 @@
-require 'nokogiri'
+require 'octokit'
 
 module GithubCommitsJob
   extend self
 
   def run
     Project.with_github_url.each do |project|
-      update_protocol(project)
       update_total_commit_count_for(project)
       update_user_commit_counts_for(project)
     end
   end
 
+
   private
 
   def update_total_commit_count_for(project)
-    begin
-      doc = Nokogiri::HTML(open(project.github_url))
-      commit_count = doc.at_css('li.commits .num').text.strip.gsub(',', '').to_i
-      project.update(commit_count: commit_count)
-      puts "Got what I came for from #{project.github_url}".green
-    rescue Exception
-      puts "I refuse to fail or be stopped by #{project.github_url}!".red
+    repo_commits = Array.new
+
+    client.contributors("#{project.github_repo_name}/#{project.github_repo_user_name}", true, per_page: 100).each do |contrib|
+      repo_commits << contrib.contributions
     end
+
+    commit_count = repo_commits.reduce(:+)
+    project.update(commit_count: commit_count)
```

Comment:
could use reduce directly here ...

```ruby
  def update_total_commit_count_for(project)
    commit_count = client.contributors(github_url(project), true, per_page: 100).reduce(0) do |memo, contrib|
      memo += contrib.contributions
    end
    project.update(commit_count: commit_count)
  end

  def github_url(project)
    "#{project.github_repo_name}/#{project.github_repo_user_name}"
  end
```
and this github_url method I've extracted could be pulled into project I think ...